### ISSUE TITLE
dgl NodeEmbedding times out too early for multi-gpu training

### ISSUE THREAD

**Issue by sojiadeshina** [dmlc/dgl#2966]
## 🐛 Bug

`dgl.nn.NodeEmbedding` supports multi cpu training on a single instance using torch.distributed. It uses the torch distributed TCPStore to set and get the embeddings in shared memory so each process can access it. The timeout is set to 30 seconds for the TCP store. https://github.com/dmlc/dgl/blob/master/python/dgl/nn/pytorch/sparse_emb.py#L92-L93. This means that the processes may not be able to get the embeddings before the call times out. 

## To Reproduce

1. Create a large graph `g` with ~250 million nodes on an instance with > 1 gpu
1. Init torch distributed data parallel with torch.distributed.init_process group.
1. Create a node embedding with dgl.nn.NodeEmbedding(g.number_of_nodes, embedding_dim=128, name='dim')

## Expected behavior
The timeout should be larger

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:
